src/util/getNextThoughtsWithContextChain.js

```diff
@@ -0,0 +1,113 @@
+import { store } from '../store.js'
+
+import {
+  ROOT_TOKEN,
+  RANKED_ROOT
+} from '../constants.js'
+// util
+import {
+  head,
+  contextOf,
+  unroot,
+  getThoughtsRanked,
+  getThoughtsSorted,
+  getSortPreference,
+  isFunction,
+  meta,
+  pathToContext,
+  getContextsSortedAndRanked,
+  hashThought,
+  splitChain,
+  nextSibling as thoughtNextSibling
+} from '../util.js'
+import { isContextViewActive } from './isContextViewActive.js'
+
+const contextToThought = (context, thoughtIndex) => context.map(contextValue => {
+  const thought = thoughtIndex[hashThought(contextValue)]
+  return { value: thought.value, rank: thought.contexts[0] ? thought.contexts[0].rank : 0 }
```

Comment:
Defaulting the rank to `0` is a red flag... seems very likely to lead to problems.